app/controllers/controllershelpers/HomeCardGenerator.scala

```diff
@@ -130,32 +114,30 @@ class HomeCardGenerator @Inject() (
   def getSelfAssessmentCard()(implicit
     messages: Messages,
     request: UserRequest[_]
-  ): Option[HtmlFormat.Appendable] = {
-    val isItsaEnrolled = enrolmentsHelper.itsaEnrolmentStatus(request.enrolments).isDefined
-    request.trustedHelper.map(_ => None).getOrElse {
-      if (isItsaEnrolled || request.isSa) {
-        if (isItsaEnrolled) {
-          Some(itsaMergeView((current.currentYear + 1).toString))
-        } else {
-          callAndContent match {
-            case Some((redirectUrl, paragraphMessageKey)) =>
-              Some(saMergeView((current.currentYear + 1).toString, redirectUrl.url, paragraphMessageKey))
-            case _                                        => None
-          }
+  ): Option[HtmlFormat.Appendable] =
+    if (request.trustedHelper.isDefined) {
+      None
+    } else {
+      val isItsaEnrolled = enrolmentsHelper.itsaEnrolmentStatus(request.enrolments).isDefined
```

Comment:
This code would be clearer I think if changes to match, i.e. (isItsaEnrolled, request.isSa, configDecorator) match

Reply:
Added match case to whereever I thought is necessary, also changed few vriable names for better redability.

Reply:
looks good